File: frontend/test_supabase.mjs
import { createClient } from '@supabase/supabase-js';

const url = "https://hkybdnbrrdjkrotwftrn.supabase.co";
const key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhreWJkbmJycmRqa3JvdHdmdHJuIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODY3Njc1NjEsImV4cCI6MjEwMjM0MzU2MX0.n6CfsNz32OH-jPK0unDY9outlr0kFHRIlwKqN6pW4ss";

const supabase = createClient(url, key);
console.log("Supabase Client Created successfully!");
const { data, error } = await supabase.from('jobs').select('*');
console.log("Data fetched from Supabase jobs table:", data, "Error:", error);

File: backend/test_supabase.py
import os
from dotenv import load_dotenv
from supabase import create_client, Client

load_dotenv()
supabase_url = os.environ.get("SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_KEY")

try:
    supabase: Client = create_client(supabase_url, supabase_key)
    print("Testing Backend Supabase Connection...")
    response = supabase.table('todos').select("*").limit(1).execute()
    print("✅ Backend Connection Successful! Fetched data:", response.data)
except Exception as e:
    if 'relation "public.todos" does not exist' in str(e) or '42P01' in str(e):
        print("✅ Backend Connection Successful! (Note: 'todos' table does not exist, but we reached the database)")
    else:
        print("❌ Backend Connection Failed:", e)
